clients/src/main/java/org/apache/kafka/common/utils/ByteBufferOutputStream.java

```diff
@@ -105,7 +105,7 @@ public int initialCapacity() {
         return initialCapacity;
     }
 
-    private void maybeExpandBuffer(int remainingRequired) {
+    public void maybeExpandBuffer(int remainingRequired) {
```

Comment:
> we expose access to the buffer directly

Are you referring to the buffer in `MemoryRecordsBuilder`? I think we are only exposing it for de-allocation for now before this PR?